### ISSUE TITLE
`ProcessingTests.ABP_When_Getting_Twin_Fails_Should_Work_On_Retry` fails sporadically on CI

### ISSUE THREAD

**Issue by spygi** [Azure/iotedge-lorawan-starterkit#772]
[See CI run](https://github.com/Azure/iotedge-lorawan-starterkit/runs/4165395779?check_suite_focus=true)

Logs:
```
Failed LoRaWan.Tests.Integration.ProcessingTests.ABP_When_Getting_Twin_Fails_Should_Work_On_Retry(deviceGatewayID: "test-gateway") [185 ms]
  Error Message:
   The collection was expected to contain a single element, but it was empty.
  Stack Trace:
     at LoRaWan.Tests.Integration.ProcessingTests.ABP_When_Getting_Twin_Fails_Should_Work_On_Retry(String deviceGatewayID) in /home/runner/work/iotedge-lorawan-starterkit/iotedge-lorawan-starterkit/Tests/Integration/ProcessingTests.cs:line 985
--- End of stack trace from previous location where exception was thrown ---
```

**Comment by bastbu** [comment]
Created #808 as a follow-up to simplify debugging of such tests in the future.

**Comment by bastbu** [comment]
As discussed with @p-schuler, we close the bug since it's not reproducible and might get addressed with #568
